pages/api/studentPortal/freemium/showModal.ts

```diff
@@ -0,0 +1,24 @@
+import { differenceInHours } from "date-fns";
+import { useSelector } from "react-redux";
+import { profileSelector } from "../../../../redux/profileSlice";
+
+
+
+export function showModal() {
+    const { userRole, userProfileData } = useSelector(profileSelector);
+    const lastSeenTime = userProfileData.lastSeen;
+    const lastSeenDateTime = new Date(lastSeenTime);
+    const currentDateTime = new Date();
+    const timeDifferenceInHours = differenceInHours(
+      currentDateTime,
+      lastSeenDateTime
+    );
+    if (
+      timeDifferenceInHours >= 24 &&
```

Comment:
this is pretty clever, I think it should work. that being said I didn't see the modal immediately and I don't think I logged into skillify over the weekend 

one thought - can you make sure the timezones for the timestamp in the client and the database line up. want to make sure one is not being store in UTC and another in local

Reply:
I see your userRole is changed to "freemium" so it should work.
As far as I can tell, they are both in UTC time --
![image](https://user-images.githubusercontent.com/111075855/226427435-54eab1aa-48e5-49e0-96d1-f74fee213df5.png)
And I did use a function to convert them to UTC time and the output remained the same...I'll keep looking into this.

Reply:
ah that makes sense...

@vithushan19 - would it be overkill to create an extra field in one of the hasura tables to keep track of this? i.e when they login and see the modal, we write to hasura that they saw it. And we read from hasura to decide whether to show the modal

not sure if there are better ways to do this

Reply:
i dont' think it's overkill, you can also create a separate table and use a foreign key into the user table